homeassistant/components/webhook/__init__.py

```diff
@@ -111,7 +127,21 @@ async def async_handle_webhook(
         _LOGGER.debug("%s", content)
         return Response(status=HTTPStatus.OK)
 
-    if webhook["local_only"]:
+    if method_name not in webhook["allowed_methods"]:
```

Comment:
I'm testing this PR - this line raises a KeyError if a previously setup webhook does not have the key 'allowed_methods'. Given the random ID and frequency of the webhook triggering this, I am guessing the companion apps.

Reply:
Thanks @hmmbob! Could you help me troubleshoot this a bit? Could you catch the exception and log the `webhook["domain"]`? That would help me a lot with tracking this down.

Reply:
Sure, here you go.
[logs.txt](https://github.com/home-assistant/core/files/9277207/logs.txt)

I have hacked my way around this - I am no dev so it might not be pretty:

```suggestion
    try:
        allowed_methods_list = webhook["allowed_methods"]
    except KeyError:
        allowed_methods_list = DEFAULT_METHODS

    if method_name not in allowed_methods_list:
```

Reply:
Thanks! There is some code in `async_register` (line 52) that should be doing this same thing. That's why I'm a bit confused. I don't understand why that code isn't working. Could you add `_LOGGER.info("%r", webhook)` just above line 130? I'd really be curious to see what that webhook looks like.

Reply:
I will, but won't be before tomorrow :+1:

Reply:
Ohh! I think I might see what is happening. From your log, it looks like this PR was installed as a custom component. Is that correct? If that is the case all of the core/built-in components, like the mobile_app, will still be using the `async_register` method from the core/built-in webhook component and won't be calling the method inside the custom component. The `allowed_methods` key definitely won't be populated under that situation and it'll result in exactly the KeyError that you are seeing.

I think this is working as intended for now. Once this PR is in the core everything will work correctly. I don't see similar workarounds for `"local_only"` or `"handler"`, so I'd rather not implement a workaround unless it is absolutely needed. Let's keep this thread unresolved though, and let the core devs weigh-in on whether they'd like a workaround added.

Reply:
You are right: I have patched the original files directly inside the container (and removed the custom_component), and the `KeyError` does not appear. I do see the app webhooks coming in into the debug logs correctly. 

I think this is resolved, as the actual intended use of the PR does not raise the KeyError - it's just my impatience and running this PR as a `custom_component` that made this error appear.